zorg/buildbot/commands/NinjaCommand.py

```diff
@@ -1,110 +1,86 @@
     haltOnFailure = True
     description = ["building"]
     descriptionDone = ["build"]
-    renderables = (
-        'options',
-        'targets',
-        'ninja',
-    )
 
-    def __init__(self, options=None, targets=None, ninja=DEFAULT_NINJA, logObserver=None, **kwargs):
+    renderables = ["options", "targets", "ninja", "jobs", "loadaverage"]
+
+    def __init__(
+        self,
+        options=None,
+        targets=None,
+        ninja=DEFAULT_NINJA,
+        logObserver=None,
+        jobs=None,
+        loadaverage=None,
+        **kwargs,
+    ):
         self.ninja = ninja
         self.targets = targets
+        self.options = options or []
 
-        if options is None:
-            self.options = list()
-        else:
-            self.options = list(options)
+        # The options are always list().
+        if isinstance(self.options, str):
+            self.options = self.options.split(" ")
+        if not isinstance(self.options, list):
+            self.options = [self.options]
 
         if logObserver:
             self.logObserver = logObserver
-            self.addLogObserver('stdio', self.logObserver)
-
-        j_opt = re.compile(r'^-j$|^-j\d+$')
-        l_opt = re.compile(r'^-l$|^-l\d+(\.(\d+)?)?$')
-
-        command = list()
-        command += [self.ninja]
-
-        # We can get jobs in the options. If so, we would use that.
-        if not any(j_opt.search(opt) for opt in self.options if isinstance(opt, str)):
-            # Otherwise let's see if we got it in the kwargs.
-            if kwargs.get('jobs', None):
-                self.options += ["-j", kwargs['jobs']]
-            else:
-                # Use the property if option was not explicitly
-                # specified.
-                command += [
-                    WithProperties("%(jobs:+-j)s"),
-                    WithProperties("%(jobs:-)s"),
-                    ]
-
-        # The same logic is for handling the loadaverage option.
-        if not any(l_opt.search(opt) for opt in self.options if isinstance(opt, str)):
-            if kwargs.get('loadaverage', None):
-                self.options += ["-l", kwargs['loadaverage']]
-            else:
-                command += [
-                    WithProperties("%(loadaverage:+-l)s"),
-                    WithProperties("%(loadaverage:-)s"),
-                    ]
+            self.addLogObserver("stdio", self.logObserver)
 
-        if self.options:
-            command += self.options
+        self.jobs = jobs
+        self.loadaverage = loadaverage
+
+        # Update the environment variables with the ninja status format settings.
+        env = kwargs.get("env") or {}
+
+        # Note: At this point env could be a dictionary or a Property object.
+        if isinstance(env, dict):
+            if "NINJA_STATUS" not in env:
+                env = env.copy()
+                env["NINJA_STATUS"] = "%e [%u/%r/%f] "
+        #else:
+            # FIXME: Ignore Property('vs_env') for now.
```

Comment:
Did you mean `# FIXME: Properly handle  Property('vs_env') here.`?

Reply:
Yes. I mixed `FIXME` and the comment for the next line with logging. Updated.